packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -385,6 +399,90 @@ export class FilesService {
         return Number(result.rows[0].row_number) - 1;
     }
 
+    async copyFilesToScope({ fileIds, targetScope }: { fileIds: string[]; targetScope: DamScopeInterface }) {
+        const files = await this.findMultipleByIds(fileIds);
+
+        if (files.length === 0) {
+            throw new Error("No valid file ids provided");
+        }
+
+        const scopeString = files[0].scope ? Object.values(files[0].scope).join("-") : "unknown";
+        const date = new Date();
+        const importFolder = await this.foldersService.create(
+            {
+                name: `Imported from ${scopeString} on ${format(date, "dd.MM.yyyy")} at ${format(date, "HH:mm:ss")}`,
+                isImportFolder: true,
+            },
+            targetScope,
+        );
+
+        let numberNewlyCopiedFiles = 0;
+        let numberAlreadyCopiedFiles = 0;
+
+        const mappedFiles: Array<{ rootFile: FileInterface; copy: FileInterface; isNewCopy: boolean }> = [];
+        for (const file of files) {
+            // The objects are cloned because file.scope is an instance of DamScope
+            // and targetScope is a plain object. Thus, they are not deep equal
+            // although they have the same scope
+            if (file.scope && isEqual({ ...file.scope }, { ...targetScope })) {
+                continue;
+            }
+
+            let copiedFile: FileInterface;
+            let isNewCopy: boolean;
+
+            const copiesInTargetScope = await this.findCopiesOfFileInScope(file.id, targetScope);
+            if (copiesInTargetScope.length > 0) {
+                copiedFile = copiesInTargetScope[0];
+                numberAlreadyCopiedFiles++;
+                isNewCopy = false;
+            } else {
+                let fileImageInput: Partial<FileImage> | undefined;
+                if (file.image) {
+                    fileImageInput = { ...Utils.copy(file.image) };
+
+                    if (fileImageInput) {
+                        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                        // @ts-ignore
+                        delete fileImageInput.id;
+                        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                        // @ts-ignore
+                        delete fileImageInput.file;
```

Comment:
Two approaches of copying the file:

1. Creating a new object and explicitly assigning all properties you want to copy
2. Copying the object and deleting the unwanted properties

I chose option 2 because I think this will prevent bugs in the future (not copying new properties of file).

Reply:
those ts-ignore are not nice....

could you use assign instead and exclude those props?